hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/hudi/command/TruncateHoodieTableCommand.scala

```diff
@@ -68,7 +71,12 @@ case class TruncateHoodieTableCommand(
       val targetPath = new Path(basePath)
       val engineContext = new HoodieSparkEngineContext(sparkSession.sparkContext)
       val fs = FSUtils.getFs(basePath, sparkSession.sparkContext.hadoopConfiguration)
+      val hoodieWriteConfig = HoodieWriteConfig.newBuilder().withPath(basePath).withProps(properties).withEngineType(EngineType.SPARK)
+        .build()
+      val transactionManager = new TransactionManager(hoodieWriteConfig, fs)
+      transactionManager.beginTransaction(org.apache.hudi.common.util.Option.empty(), org.apache.hudi.common.util.Option.empty())
       FSUtils.deleteDir(engineContext, fs, targetPath, sparkSession.sparkContext.defaultParallelism)
+      transactionManager.endTransaction(org.apache.hudi.common.util.Option.empty())
```

Comment:
Can the OCC detect the conflicts of truncate operations?

Reply:
It cannot be intercepted. Truncate table has no transaction control when there is no partition.

Reply:
Then what the purpose of the transaction manager?

Reply:
When the truncate table and the insert statement are executed at the same time, the table data may be damaged.

Reply:
But between the `beginTransaction` and `endTransaction`, we only delete some files from the table, there is no new commit metadata genetated, so how the txn manager can help here(detect the conflicts)?

Reply:
This is at the table level, so there should be no need to pass the newTxnOwnerInstant parameter.

Reply:
Sorry, I'm confused.

Reply:
Guess you are utilizating the distributed lock for gurading the sequence right?

Reply:
Can we use replace commit here to provide the transaction mechanism? We have already done it for partitioned table

Reply:
+1 on using replace commit. This will be truly revertible and aligns with other operations. @waywtdcc : Can you make this change ?